### PR TITLE
#7246 Do not export styles when map is exported

### PR DESCRIPTION
**Pull request by alexander-fedorenko**
## Description
Export to MapStore format will not contain "availableStyles" array regardless if it exists in current map state or not.

Layer's property "availableStyles" is not in the store by default when layer is added to the map. It's getting written into the store with action "UPDATE_NODE" as a result of running "getAvailableStylesFromLayerCapabilities" from "epics/styleeditor.js" when user opens layer settings:
image

So if user run export when layer settings was never open - there are no availableStyles in exported data when MapStore format is used.

This PR changes such behavior to exclude availableStyles export explicitly, regardless if they were loaded into the store or not.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7246 

**What is the new behavior?**
- MapStore export format - "availableStyles" property won't be exported regardless if it's in store or not.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/epics/mapexport.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ const PersistMap = {
         }
 
         return Rx.Observable.forkJoin(...layers.map(layer => getLayerCapabilities(layer).catch(() => Rx.Observable.of(null))))
+            .switchMap(capArr => capArr.map(({ style, ...cap}) => cap))
```
Comment (allyoucanmap) [inline web/client/epics/mapexport.js:62]: Hi @alexander-fedorenko , I see this is an export for WMC format and I'm not sure the issue wants to target that kind of export. I think the WMC export should be kept as is now.
@tdipisa could you please confirm this?

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/epics/mapexport.js (hunk 1 above)
Comment (tdipisa) [reply web/client/epics/mapexport.js:62]: WMC Export is not involved at all here.

Comment (allyoucanmap) [reply web/client/epics/mapexport.js:62]: @alexander-fedorenko please remove the fix you introduced for the WMC export. The other files seem good

Comment (alexander-fedorenko) [reply web/client/epics/mapexport.js]: @allyoucanmap @tdipisa Changes in WMC export are rolled-back now. PR description is updated as well.

**Review by allyoucanmap — APPROVED** [review]

**Comment by tdipisa** [comment]
@ElenaGallo please test this well in DEV and let us know if it is possible to proceed with the backport.

**Comment by ElenaGallo** [comment]
Test passed, @alexander-fedorenko please backport to 2021.02.xx. Thanks

**Pull request merged**